mbuild/packing.py

```diff
@@ -176,27 +175,31 @@ def fill_box(compound, n_compounds=None, box=None, density=None, overlap=0.2,
     box_maxs -= edge * 10
 
     # Build the input file for each compound and call packmol.
-    filled_pdb = tempfile.mkstemp(suffix='.pdb')[1]
-    input_text = PACKMOL_HEADER.format(overlap, filled_pdb, seed)
-
-    for comp, m_compounds, rotate in zip(compound, n_compounds, fix_orientation):
-        m_compounds = int(m_compounds)
-        compound_pdb = tempfile.mkstemp(suffix='.pdb')[1]
-        comp.save(compound_pdb, overwrite=True)
-        input_text += PACKMOL_BOX.format(compound_pdb, m_compounds,
-                           box_mins[0], box_mins[1], box_mins[2],
-                           box_maxs[0], box_maxs[1], box_maxs[2],
-                           PACKMOL_CONSTRAIN if rotate else "")
-
-    _run_packmol(input_text, filled_pdb, temp_file)
-
-    # Create the topology and update the coordinates.
-    filled = Compound()
-    for comp, m_compounds in zip(compound, n_compounds):
-        for _ in range(m_compounds):
-            filled.add(clone(comp))
-    filled.update_coordinates(filled_pdb)
-    filled.periodicity = np.asarray(box.lengths, dtype=np.float32)
+    filled_pdb = _new_pdb_file()
+    try:
+        input_text = PACKMOL_HEADER.format(overlap, filled_pdb.name, seed)
+        for comp, m_compounds, rotate in zip(compound, n_compounds, fix_orientation):
+            m_compounds = int(m_compounds)
+            compound_pdb = _new_pdb_file()
+            comp.save(compound_pdb.name, overwrite=True)
```

Comment:
I think after this line, Python is done with the instances of `compound_pdb`. Packmol itself will read it but I think it could be closed here.

Reply:
I tried closing those files within the for loop before, but since `input_text` requires the `compound_pdb` file path: `input_text += PACKMOL_BOX.format(compound_pdb.name, m_compounds,...`

Those files have to stay open until after `run_packmol` has been called.

Reply:
Correct me if I'm wrong, but won't the file still exist after Python closes it? Or will it be deleted due to it being a temporary file? Or is it simply a matter of losing track of the _name_ of the file?

Reply:
Yeah, you are completely right. I realize I was trying to close **and** delete the files before. Closing here is completely fine, since packmol is able to open the file afterwards.